### PR TITLE
Thread local allocation of typed objects

### PR DESCRIPTION
**Pull request by tholp**

### CODE REVIEW

**Comment by tholp** [comment]
This will make typed (and typed array) object allocation use the thread local feature.

(The request includes "Missing voilatile declaration")

File: include/private/gc_priv.h [hunk 1]
```diff
@@ -1641,7 +1641,7 @@ GC_EXTERN void (*GC_push_typed_structures)(void);
                         /* the typed allocation support if unused.      */
 
 GC_INNER void GC_with_callee_saves_pushed(void (*fn)(ptr_t, void *),
-                                          ptr_t arg);
+                                          volatile ptr_t arg);
```
Comment (ivmai) [inline include/private/gc_priv.h]: Why do we need volatile here? Is there a compiler complaining about it? (Even the definition has it, it is meaningless in the declaration

Comment (tholp) [inline include/private/gc_priv.h]: Yes, Microsoft Visual C++ (with Warning Level 4) gives a warning.

In addition we use "Treat Warnings as Errors" so for us it is an error.

Comment (ivmai) [inline include/private/gc_priv.h]: OK, merged to master - 65b50c0

**Pull request closed without merging**